sopel_unicode/plugin.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import annotations
+import logging
+from typing import Iterable
+
+from sopel import plugin
+from sopel.config import types
+
+from .impl import NAME_TO_CODEPOINT, describe_char, unicodedata
+
+
+logger = logging.getLogger(__name__)
+
+PREFIX = plugin.output_prefix('[unicode] ')
+
+
+class SopelUnicodeSection(types.StaticSection):
+    max_length = types.ValidatedAttribute('max_length', parse=int, default=5)
+    """Maximum length of Unicode string input"""
+    length_override_channels = types.ListAttribute('length_override_channels')
+    """Channels where max_length does not apply"""
+    ignore_characters = types.ListAttribute('ignore_characters', default=[" "])
+    """Characters ignored during lookup"""
+    search_max_matches = types.ValidatedAttribute('search_max_matches', parse=int, default=10)
+    """Maximum number of matches for a codepoint search"""
+    search_num_public_matches = types.ValidatedAttribute('search_num_public_matches', parse=int, default=2)
+    """Number of matches publicly reported for a codepoint search"""
+
+
+def configure(config):
+    config.define_section('sopel-unicode', SopelUnicodeSection)
+    config.foo.configure_setting('max_length', 'How many codepoints maximum in an lookup string?')
+    config.foo.configure_setting('length_override_channels', 'Which channels should bypass the maximum input length? Leave blank for none.')
+
+
+def setup(bot):
+    bot.settings.define_section('sopel_unicode', SopelUnicodeSection)
```

Comment:
You are defining the config section with two different names. The hyphenated form is not a good idea since the section name is also used as a Python attribute name. Usually a plugin's config section is named "plugin_name".

Also note that `configure()` is referring to a `foo` section. I'd suggest something like this:

```suggestion
def configure(config):
    config.define_section('unicode_info', SopelUnicodeSection)
    config.unicode_info.configure_setting('max_length', 'How many codepoints maximum in an lookup string?')
    config.unicode_info.configure_setting('length_override_channels', 'Which channels should bypass the maximum input length? Leave blank for none.')
    # any other settings you think would be useful to have in the wizard

def setup(bot):
    bot.settings.define_section('unicode_info', SopelUnicodeSection)
```

But it might be more work than it's worth to change the rest of the file's references to `bot.settings.sopel_unicode` 🤷‍♂️

Reply:
Should be fixed by 8abc8ba, thanks